### PR TITLE
feat: add teacher config/sft loss for hosted training SFT

### PR DESCRIPTION
**Pull request by eligotts**
## Summary

Implements APR-157 SFT distillation support through the existing `prime train` config path.

- Adds top-level `loss = "rl" | "sft"` with `[teacher]` and `[teacher.sampling]`.
- Builds teacher sampling on a general sampling-args config so provider/API-specific fields can pass through without adding new CLI schema for every parameter.
- Validates that SFT requires a teacher, RL rejects teacher config, and `teacher.save = true` is not supported.
- Defaults omitted SFT `rollouts_per_example` to `1` while preserving explicit overrides.
- Forwards the public platform payload shape as `loss` and `teacher`.
- Routes top-level `enable_thinking` and `reasoning_effort` sampling aliases into `extra_body.chat_template_kwargs`.
- Updates the confirmation summary so `Training`, `Teacher`, and `Run Config` render as separate sections.
- Keeps the generated template TOML-safe when SFT teacher config and `checkpoint_id` are uncommented together.

## Config Shape

```toml
model = "openai/gpt-oss-20b"
loss = "sft"

[teacher]
model = "openai/gpt-oss-120b"
save = false

[teacher.sampling]
max_tokens = 2048
enable_thinking = false
reasoning_effort = "medium"

[[env]]
id = "primeintellect/wordle"
```

## API Payload

```json
{
  "loss": "sft",
  "teacher": {
    "model": "openai/gpt-oss-120b",
    "save": false,
    "sampling": {
      "max_tokens": 2048,
      "extra_body": {
        "chat_template_kwargs": {
          "enable_thinking": false,
          "reasoning_effort": "medium"
        }
      }
    }
  }
}
```

## Tests

- `uv run pytest packages/prime/tests/test_rl_config.py packages/prime/tests/test_train_cli.py -q`

### CODE REVIEW

**Comment by tim0120** [comment]
Opened a newer APR-157 draft using the GA-facing `loss` / `[teacher]` config shape from the current hosted-training notes: https://github.com/PrimeIntellect-ai/prime/pull/679. I would treat #679 as the successor to this older `loss_type` / `teacher_rollout_model` version unless we decide to keep the old API shape.
